app/experimenter/nimbus-ui/src/hooks/useChangeOperationMutation.test.tsx

```diff
@@ -20,28 +19,65 @@ import { useChangeOperationMutation } from "./useChangeOperationMutation";
 
 describe("hooks/useChangeOperationMutation", () => {
   it("can successfully execute mutation set callbacks", async () => {
-    const { callbacks, submitError } = setupTestHook();
+    const { result, invokeCallback, waitForNextUpdate } = setupTestHook();
 
-    for (const callback of callbacks) {
-      await act(async () => void callback());
-      await waitFor(() => expect(submitError).toBeNull());
+    const callbackCount = result.current.callbacks.length;
+    for (let idx = 0; idx < callbackCount; idx++) {
+      invokeCallback(idx);
+      await waitForNextUpdate();
+      expect(result.current.submitError).toBeNull();
     }
   });
 
-  it("indicates when loading", async () => {
-    const {
-      isLoading,
-      callbacks: [callback],
-    } = setupTestHook();
-    expect(isLoading).toBeFalsy();
-    await act(async () => void callback());
-    waitFor(() => expect(isLoading).toBeTruthy());
+  it("indicates when loading mutation", async () => {
```

Comment:
Reworking these tests to use [`await waitForNextUpdate()`](https://react-hooks-testing-library.com/usage/advanced-hooks#async) seems to exercise the functionality better. But I suspect there might be an intermittent race condition in here still that I haven't been able to flush out

Reply:
Circle reveals all.